Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -75,7 +76,7 @@ def run
             end
           end
 
-          parallel = false if args.profile
+          parallel = false if args.profile || (args.coverage? && files.length < Hardware::CPU.cores)
```

Comment:
```suggestion
          parallel = false if args.profile || args.only
```
would be simpler and I think would actually be pretty reasonable

Reply:
While conceptually simpler, `--only` isn't strictly an indicator that a `brew tests` run is small. For example, `--only=cask` involves 45 specs and takes ~54 seconds on my machine with parallel execution (14 processes on an M4 Pro) vs. ~3.5 minutes with `parallel = false`. `--only` also takes a comma-separated list, so that can lead to runs with a larger number of specs as well.

Besides that, I believe we only need a workaround when `--coverage` is used, so disabling parallel execution for non-coverage runs would unnecessarily slow things down.

Though my approach in this PR isn't as simple, I've tried to limit this workaround to only when it's needed.

Reply:
I meant to add an explanatory comment but kept forgetting, so thanks for the reminder. Added now 👍